web/base_pdf_page_view.js

```diff
@@ -100,10 +112,41 @@ class BasePDFPageView {
     // have a final flash we just display it once all the drawing is done.
     const updateOnFirstShow = !prevCanvas && !hasHCM && !hideUntilComplete;
 
-    const canvas = (this.canvas = document.createElement("canvas"));
+    let canvas = (this.canvas = document.createElement("canvas"));
 
     this.#showCanvas = isLastShow => {
+      let tempCanvas = this.#tempCanvas;
       if (updateOnFirstShow) {
+        if (!isLastShow && this.#minDurationToUpdateCanvas > 0) {
+          // We draw on the canvas at 60fps (in using `requestAnimationFrame`),
+          // so if the canvas is large, updating it at 60fps can be a way too
+          // much and can cause some serious performance issues.
+          // To avoid that we only update the canvas every
+          // minDurationToUpdateCanvas ms.
```

Comment:
Nit: To account for this now longer being a code constant (as in first version of the patch).
```suggestion
          // To avoid that we only update the canvas every
          // `this.#minDurationToUpdateCanvas` ms.
```